### PR TITLE
Fix incorrect parsing of string contains "verus!{".

### PR DESCRIPTION
**Pull request by XuhengLi**
non_verus is defined to accept ANY which might be of low precedency. When a string literal in non_verus contains "verus!{", VerusParser fails to parse the string and throws an error indicating the string is parsed into verus_macro_use.

This patch modfies non_verus so it accepts string, raw_string, etc., explicitly so string literals with "verus!{" can be correctly parsed.

verus-minimal is changed as well otherwise rustfmt shipped in this crate cannot format the non-verus code correctly.

A companion unit test which should fail without this patch is included.

### CODE REVIEW

File: src/verus.pest
Comment (jaybosamiya) [inline src/verus.pest:76]: Minor suggestion (you should be able to accept this from the GitHub UI)
```suggestion
/// happens, then we need to make sure that we aren't accidentally starting
/// parsing in the middle of the macro_rules macro body itself. Similarly, strings.
non_verus = @{
```

**Review by jaybosamiya — APPROVED** [review]
Awesome, thanks for fixing this issue!

This overall looks good to me, modulo the minor comment about the comment which would be nice to add in before we merge this.

Majority of the addition to `verus-minimal.pest` is copying over `string`,...`ray_byte_string` into it; just thinking about longer term maintainability (as a TODO item for myself to take care of after this PR; no changes needed here) is to add some comments on which things need to stay in sync.

**Comment by XuhengLi** [comment]
Formatting updates.

**Comment by jaybosamiya** [comment]
Overall, looks good to me. Thanks for your first contribution to verusfmt! Will merge it in once CI goes green.

**Pull request merged**
